compiler/rustc_typeck/src/impl_wf_check/min_specialization.rs

```diff
@@ -384,9 +389,17 @@ fn check_specialization_on<'tcx>(tcx: TyCtxt<'tcx>, predicate: ty::Predicate<'tc
                     .emit();
             }
         }
+        ty::PredicateKind::Projection(ty::ProjectionPredicate { projection_ty, term }) => {
+            tcx.sess
+                .struct_span_err(
+                    span,
+                    &format!("cannot specialize on associated type `{projection_ty} == {term}`",),
```

Comment:
hm, i guess i could rebind `projection_ty`, but i don't think people are specializing on `for<'a> Ty: Trait<Proj = &'a ()>`, so it's not necessarily worth the complexity I guess. Also we'd need to skip the binder on term still, so we don't print `for<..> assoc == for<..> term`..

Reply:
yea, not really worth it for the diagnostic. You could emit another note if the binders actually bind anything stating that specialization on lifetimes is not... ideal. But unless we have some good specialization docs for that that we could link to, not really worth it.